### DISCUSSION TITLE
z2m Service restaring all the time

### DISCUSSION THREAD

**Discussion by wwyyrrmm** [Koenkk/zigbee2mqtt#22849] [Show and tell]
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         z2m: Received Zigbee message from 'BB EG Gaestebad', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":7740}' from endpoint 1 with groupID 0
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         z2m: MQTT publish: topic 'zigbee2mqtt/BB EG Gaestebad', payload '{"battery":60,"humidity":77.4,"last_seen":"2024-05-31T13:12:45+02:00","linkquality":156,"temperature":20.7}'
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: <-- [254,1,100,1,0,100]
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --- parseNext [254,1,100,1,0,100]
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --> parsed 1 - 3 - 4 - 1 - [0] - 100
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:znp: SRSP: <-- AF - dataRequest - {"status":0}
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --- parseNext []
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: <-- [254,3,68,128,0,1,10,204]
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --- parseNext [254,3,68,128,0,1,10,204]
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --> parsed 3 - 2 - 4 - 128 - [0,1,10] - 204
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:znp: AREQ: <-- AF - dataConfirm - {"status":0,"endpoint":1,"transid":10}
May 31 13:12:45 raspberrypi node[563973]: [2024-05-31 13:12:45] debug:         zh:zstack:unpi:parser: --- parseNext []
May 31 13:12:51 raspberrypi systemd[1]: **_zigbee2mqtt.service: start operation timed out. Terminating._**
May 31 13:12:51 raspberrypi node[563973]: [2024-05-31 13:12:51] debug:         z2m: Saving state to file /opt/zigbee2mqtt/data/state.json
May 31 13:12:51 raspberrypi node[563973]: [2024-05-31 13:12:51] debug:         z2m: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload '{"state":"offline"}'
May 31 13:12:51 raspberrypi node[563973]: [2024-05-31 13:12:51] info:         z2m: Disconnecting from MQTT server
May 31 13:12:51 raspberrypi node[563973]: [2024-05-31 13:12:51] info:         z2m: Stopping zigbee-herdsman...

I do not know why z2m is rrestarting ervery minute (2Minutes) the devices reporting values before are also not unique.

Zigbee2MQTT version
    [1.37.1](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.37.1) commit: [cda867a](https://github.com/Koenkk/zigbee2mqtt/commit/cda867a)

Coordinator type
    zStack3x0

Coordinator revision
    20230507

Coordinator IEEE Address
0x........

Frontend version
    0.6.167

    19.37.2

    0.46.6

Stats

        Total 39
        By device type
        End devices: 32
        Router: 7
        By power source
        Battery: 32
        Mains (single phase): 5
        DC Source: 2
        By vendor
        eWeLink: 18
        SONOFF: 12
        IKEA of Sweden: 6
        Paulmann Licht: 2
        LUMI: 1
        By model
        DS01: 16
        SNZB-02D: 12
        TRADFRI Driver 30W: 5
        MS01: 2
        500.45: 2
        lumi.sensor_motion: 1
        TRADFRI on/off switch: 1

Can somebody help?
Carsten

P.S.: if I run in /opt/zigbee2mqtt/
sudo npm start

z2m is running without problems

**Comment by grumpypompey** [comment]
Hi - were you able to find any further information about this?

I have set up a new installation of zigbee2mqtt on a linux VM hosted on proxmox. I am running the same version 1.37.1 commit cda867a and seeing the same continual restarting of z2m for no apparent reason. I am running a Sonoff ZBDongle-P which had an older firmware but I flashed it so it's now running the same 20230507 version.

After spending 3 days trying to figure out the problem I came across this message and stopped the service, moved to /opt/zigbee2mqtt and ran sudo npm start. So far I have 32 minutes of uptime without any restart when previously it was restarting every 1 to 2 minutes non stop.

**Comment by francisp2** [comment]
It usually is a permissions problem on the Zigbee2MQTT directory.

**Reply by wwyyrrmm** [reply]
Not in my case

**Reply by grumpypompey** [reply]
Hi,

I have a user called z2m. T

After reading another comment somewhere in these threads, I commented out the following lines from /etc/systemd/system/zigbee2mqtt.service and now when I enable the service, zigbee2mqtt starts and runs no problem - no more continuous stopping & starting:

#Type=notify
#WatchdogSec=10s

User z2m is a member of the groups "z2m" and "sudo".

The zigbee2mqtt folder has owner as "z2m" and group as "z2m". The folder permissions are: drwxr-xr-x
The file /etc/systemd/system/zigbee2mqtt.service has owner "root" and group "root". File permissions are: -rw-r--r--

Do you still think this is a permissions issue? If so then I assume it's the service file not the z2m directory? Do you know what the permissions should be for this file? And any clue as to why permissions would no longer be a problem with those lines commented out of the .service file?


Thanks :)

**Comment by wwyyrrmm** [comment]
This was my solution:
https://github.com/Koenkk/zigbee2mqtt/issues/21463
